src/components/Proposal/Create/PluginForms/CreateTokenTradeProposal.tsx

```diff
@@ -0,0 +1,325 @@
+
+import * as React from "react";
+import { connect } from "react-redux";
+import { IPluginState, IProposalCreateOptionsTokenTrade } from "@daostack/arc.js";
+import { enableWalletProvider } from "arc";
+
+import { ErrorMessage, Field, Form, Formik, FormikProps } from "formik";
+
+import { IRootState } from "reducers";
+import { NotificationStatus, showNotification } from "reducers/notifications";
+import * as arcActions from "actions/arcActions";
+
+import Analytics from "lib/analytics";
+import { isValidUrl, supportedTokens } from "lib/util";
+import { exportUrl, importUrlValues } from "lib/proposalUtils";
+
+import TagsSelector from "components/Proposal/Create/PluginForms/TagsSelector";
+import TrainingTooltip from "components/Shared/TrainingTooltip";
+import * as css from "../CreateProposal.scss";
+import MarkdownField from "./MarkdownField";
+import HelpButton from "components/Shared/HelpButton";
+import { SelectField } from "./CreateContributionRewardProposal";
+
+interface IExternalProps {
+  daoAvatarAddress: string;
+  handleClose: () => any;
+  pluginState: IPluginState;
+}
+
+const mapStateToProps = (state: IRootState, ownProps: IExternalProps) => {
+  return ownProps;
+};
+
+interface IDispatchProps {
+  createProposal: typeof arcActions.createProposal;
+  showNotification: typeof showNotification;
+}
+
+const mapDispatchToProps = {
+  createProposal: arcActions.createProposal,
+  showNotification,
+};
+
+type IProps = IExternalProps & IDispatchProps;
+
+interface IFormValues {
+  description: string;
+  title: string;
+  url: string;
+  tags: Array<string>;
+  sendTokenAddress: string,
+  sendTokenAmount: number,
+  receiveTokenAddress: string,
+  receiveTokenAmount: number,
+}
+
+interface IState {
+  tags: Array<string>;
+}
+
+class CreateKnownPluginProposal extends React.Component<IProps, IState> {
+
+  initialFormValues: IFormValues;
+
+  constructor(props: IProps) {
+    super(props);
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+
+    this.initialFormValues = importUrlValues<IFormValues>({
```

Comment:
Please convert this and other bits to use the `IFormModalService` (see how all of the other create proposal modules do it)